In [1]:

# Imports 
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

/Users/fatimaenriquez/opt/anaconda3/lib/python3.9/site-packages/scipy/__init__.py:155: UserWarning: A NumPy version >=1.18.5 and <1.25.0 is required for this version of SciPy (detected version 1.26.4
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion}"


In [2]:

df = pd.read_csv('dataset/heart.csv')
df.shape

(303, 14)

In [4]:
df.head()

age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
0   63    1   3     145   233    1        0       150     0      2.3    0   
1   37    1   2     130   250    0        1       187     0      3.5    0   
2   41    0   1     130   204    0        0       172     0      1.4    2   
3   56    1   1     120   236    0        1       178     0      0.8    2   
4   57    0   0     120   354    0        1       163     1      0.6    2   

   caa  thall  output  
0    0      1       1  
1    0      2       1  
2    0      2       1  
3    0      2       1  
4    0      2       1

What each column represents according to the author on kaggle: 
- age (age of the individual)
- sex (sex (f/m) of the individual)
- trtbps (resting blood pressure (in mm Hg))
- chol (cholestoral in mg/dl fetched via BMI sensor)
- fbs (fasting blood sugar > 120 mg/dl) (1 = true; 0 = false)
- restecg (resting electrocardiographic results)
- thalachh (maximum heart rate achieved)
- exng (exercise induced angina (1 = yes; 0 = no))
- oldpeak (previous peak)

In [19]:
age_mean = df['age'].mean()
low_age = df['age'].min()
max_age = df['age'].max()

print("The average age of the heart csv data: ", age_mean)
print("The lowest age of the heart csv data: ", low_age)
print("The max age of the heart csv data: ", max_age)

The average age of the heart csv data:  54.366336633663366
The lowest age of the heart csv data:  29
The max age of the heart csv data:  77


Below we can observe that there are 207 patients who are male and 96 who are female. However, I could not find where it stated that 0 is for females and 1 is for males. This is based on a question asked from a discussion in kaggle.

In [16]:
sex = df['sex'].value_counts()
sex

1    207
0     96
Name: sex, dtype: int64

"Mean serum total cholesterol level for adults ages 20 and older: 191 mg/dL (2015-2018)" This is according to the CDC 
- https://www.cdc.gov/nchs/fastats/cholesterol.htm

The data expreses that the cholesterol levels are in mg/dl which are fetched via BMI sensor.

In [18]:
avg_chol= df['chol'].mean()
print("The average cholestorol levels of the heart csv data: ", avg_chol)

The average cholestorol levels of the heart csv data:  246.26402640264027


Accoridng to the CDC: Before a meal an individudlas fasting blood sugar levels should be between 80 to 130 mg/dL.
- https://www.cdc.gov/diabetes/treatment/index.html#:~:text=Your%20blood%20sugar%20target&text=These%20are%20typical%20targets%3A,Less%20than%20180%20mg%2FdL.



Thus we have most individuals from the data have healthier fasting blood sugar levels.

In [25]:
sex = df['fbs'].value_counts()
sex

0    258
1     45
Name: fbs, dtype: int64